### PR TITLE
GAP-2096: change answer from summary before submit

### PR DESCRIPTION
**Pull request by ConorFayleTCO**
## Description

[GAP-2096](https://technologyprogramme.atlassian.net/browse/GAP-2096)

Summary of the changes and the related issue. List any dependencies that are required for this change:

Allow for editing the questions from the summary screen and navigating back to summary once finished, bypassing the "Have you completed this section". Add routing for back buttons and redirects using query params

Required:
https://github.com/cabinetoffice/gap-find-applicant-backend/pull/88

## Type of change

Please check the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes:

- [ ] Unit Test

- [ ] Integration Test (if applicable)

- [ ] End to End Test (if applicable)

## Screenshots (if appropriate):

Please attach screenshots of the change if it is a UI change:
Video in ticket

# Checklist:

- [ ] If I have listed dependencies above, I have ensured that they are present in the target branch.
- [x] I have performed a self-review of my code.
- [x] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation where applicable.
- [ ] I have ran cypress tests and they all pass locally.

[GAP-2096]: https://technologyprogramme.atlassian.net/browse/GAP-2096?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by jgunnCO** [comment]
applicant build failing with a ts error 

image

File: packages/applicant/src/pages/submissions/[submissionId]/summary.page.test.tsx [hunk 1]
```diff
@@ -23,6 +41,244 @@ jest.mock(
...
+
+describe('getServerSideProps', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+  it('should return a redirect to grant-is-closed when submission is REMOVED ', async () => {
+    (getApplicationStatusBySchemeId as jest.Mock).mockResolvedValue('REMOVED');
+    (getSubmissionById as jest.Mock).mockReturnValue(propsWithAllValues);
+    (getJwtFromCookies as jest.Mock).mockReturnValue('testJwt');
+    (hasSubmissionBeenSubmitted as jest.Mock).mockReturnValue(false);
```
Comment (jgunnCO) [inline packages/applicant/src/pages/submissions/[submissionId]/summary.page.test.tsx:198]: You can use the `jest.mocked` utility to get a properly typed mock function and avoid having cast these multiple times

```
const mockGetApplicationStatusByShchemeId = jest.mocked(getApplicationStatusBySchemeId;
etc
```

File: packages/applicant/src/utils/postQuestion.ts [hunk 2]
```diff
@@ -250,6 +267,7 @@ export const createRequestBody = (
     submissionId,
     questionId,
     multiResponse: isResponseAnArray ? body[questionId] : multiResponseValues,
+    shouldUpdateSectionStatus: !shouldRedirectToSummary,
```
Comment (jgunnCO) [inline packages/applicant/src/utils/postQuestion.ts:270]: Just to check: this is so we don't set complete sections to in progress when the user edits an answer?

Comment (ConorFayleTCO) [reply packages/applicant/src/utils/postQuestion.ts:270]: yessir

**Review by jgunnCO — APPROVED** [review]

**Review by jgunnCO — APPROVED** [review]

**Pull request merged**